src/js/components/chart/utils-normalization.js

```diff
@@ -0,0 +1,219 @@
+   * @param granularity the interval of the data points, uses '1' as default
+   */
+
+  getXAxis(series, granularity) {
+    let flat = series;
+    // Check whether the series has more than one set of values
+    if (Array.isArray(series[0][0])) {
+      flat = flatten(series);
+    }
+
+    const xValues = this.getXValues(flat);
+    const max = this.getMaxVal(xValues);
+    const min = this.getMinVal(xValues);
+
+    if (!granularity) {
+      // Possible to implement a calculator that with a given data points
+      // will calculate the optimized granularity
+      granularity = 1;
+      console.warn('granularity was not defined for getXAxis calculation');
+    }
+
+    let normalizedXAxisLength = ( (max - min) / granularity) + 1;
+    if (!isInteger(normalizedXAxisLength)) {
+      console.warn('X values are not corresponding to the given granularity');
+      normalizedXAxisLength = Math.floor(normalizedXAxisLength);
+    }
+
+    const normalizedXAxis = Array.apply(null, new Array(normalizedXAxisLength));
+    return normalizedXAxis.map((x, i) => i * granularity + min);
+  };
+
+  /**
+   * Getting the max value in an array
+   *
+   * @param arr
+   * return a numeric value
+   */
+  getMinVal (arr) {
+    return Math.min.apply(Math, arr);
+  }
+
+  /**
+   * Getting the min value in an array
+   *
+   * @param arr
+   * return a numeric value
+   */
+  getMaxVal (arr) {
+    return Math.max.apply(Math, arr);
+  }
+
+  /**
+   * normalize func will normalize the values in case of gaps and blank areas,
+   * and fill 'undefined' for the missing points added during normalization
+   * @param Series array of 2D arrays
+   * @param granularity the interval of the data points,
+   * uses '1' as default in case granularity is missing
+   * @returns {{}}
+   *
+   * Assumptions:
+   *  Arrays are sorted by x-values
+   *  xValue will not appear more than once in a series
+   *
+   * Example 1 (simple):
+   *
+   * Series = [
+   *           [[1,2],[3,3]] //series
+   *           [[1,4],[6,7]]
+   *          ]
+   * xAxis = [1,2,3,4,5,6]
+   * After Normalization = [
+   *                        [2, u, 3, u, u, u],
+   *                        [4, u, u, u, u, 7]
+   *                       ]
+   *
+   * Example 2:
+   *
+   * Series = [
+   *           [[1,2],[3,3],[4,5]], //series
+   *           [[1,4],[6,7],[8,9]]
+   *          ]
+   *
+   * xAxis = [1,2,3,4,5,6,7,8]
+   * After Normalization = [
+   *                        [2, u, 3, 5, u, u, u, u],
+   *                        [4, u, u, u, u, 7, u, 9]
+   *                       ]
+   *
+   * Example 3:
+   *
+   * Series = [
+   *           [[1,2],[3,3],[4,5]], //series
+   *           [[0,4],[6,7]]
+   *          ]
+   *
+   * xAxis = [0,1,2,3,4,5,6]
+   * After Normalization = [
+
+   *                        [u, 2, u, 3, 5, u, u],
+   *                        [4, u, u, u, u, u, 7]
+   *                       ]
+   *
+   */
+
+  normalize(Series, granularity) {
```

Comment:
I used uppercase only because I wanted to distinguish between multiple and single series. 
Short anecdote, It will sound funny, but in Hebrew we use different words for single series and multiple , and the only distinguish I found with google translate is that the multiple was upper case. 
Since I use the lowercase 'series' as well during the calculation, any suggestions how to call 'Series' instead?